src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.cpp

```diff
@@ -0,0 +1,192 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <gsl/gsl_sf_bessel.h>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/Integration/GslQuadAdaptive.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/IsotropicHomogeneous.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace Elasticity::Solutions {
+
+HalfSpaceMirror::HalfSpaceMirror(
+    const double beam_width, constitutive_relation_type constitutive_relation,
+    const size_t integration_intervals, const double absolute_tolerance,
+    const double relative_tolerance) noexcept
+    : beam_width_(beam_width),
+      constitutive_relation_(std::move(constitutive_relation)),
+      integration_intervals_(integration_intervals),
+      absolute_tolerance_(absolute_tolerance),
+      relative_tolerance_(relative_tolerance) {}
+
+tuples::TaggedTuple<Tags::Displacement<3>> HalfSpaceMirror::variables(
+    const tnsr::I<DataVector, 3>& x,
+    tmpl::list<Tags::Displacement<3>> /*meta*/) const noexcept {
+  const double shear_modulus = constitutive_relation_.shear_modulus();
+  const double lame_parameter = constitutive_relation_.lame_parameter();
+  auto result = make_with_value<tnsr::I<DataVector, 3>>(x, 0.);
+  const auto radius = sqrt(square(get<0>(x)) + square(get<1>(x)));
+
+  const integration::GslQuadAdaptive<
+      integration::GslIntegralType::UpperBoundaryInfinite>
+      integration{integration_intervals_};
+  const double lower_boundary = 0.;
+  const size_t num_points = get<0>(x).size();
+  for (size_t i = 0; i < num_points; i++) {
+    const double z = get<2>(x)[i];
+    const double r = radius[i];
+    if (r >= 100 * std::numeric_limits<double>::epsilon()) {
+      const double surface_term_r = 1. - (lame_parameter + 2. * shear_modulus) /
+                                             (lame_parameter + shear_modulus);
+      const double result_r =
+          0.25 / (M_PI * shear_modulus * r) *
```

Comment:
Sorry about that. This term comes from the projection on the cartesian grid: cos phi = x/r. I put the r there, because both x and y component are divided by r. Do you think it's worth doing it separately or maybe adding a comment here?

Reply:
I have removed the additional division by r, called it `displacement_r` and later multiply this by `cos_phi = x/r` and `sin_phi = y/r`.